### PR TITLE
Remove vertical bar and snap to window ability

### LINKED ISSUE #3051: You can add too many external tools to the bar
### Steps to reproduce the bug

Add lots of external tools, and pin them to the bar

### Expected result

Some limit on the number of tools, based on the size of the bar.
Also some "overflow" glyph, which the user can invoke to see the overflow items.

### Actual result

You can keep adding tools without limit, and beyond the point where they can all be shown on the bar. The excess tools are simply clipped, but only at the end of the bar - which means that tools at the end of the bar occupy the same space as the system resource usage reports.

too-many-tools

### LINKED ISSUE #3320: In vertical collapsed mode, there's no minimize or maximize button
### Dev Home version

_No response_

### Windows build number

_No response_

### Other software

_No response_

### Steps to reproduce the bug

The minimize, maximize/restore and close chrome buttons are standard. The user would legitimately expect to be able to access these 3 controls regardless of horizontal/vertical orientation. Forcing them to switch to horizontal or to expanded mode just to do minimize or maximize seems bad.

Also, the X close button seems to be bigger than the other buttons.

### Expected result

_No response_

### Actual result

Min/max not available in vertical mode

### Included System Information

_No response_

### Included Extensions Information

_No response_

### LINKED ISSUE #3701: Remove collapsed vertical mode
### Suggested new feature or improvement

Per user feedback, this is not particularly useful, and it has a number of problems.

### Scenario

Remove the feature altogether.

### Additional details

Problems include:
- In vertical orientation, we can’t render the standard chrome in the standard horizontal layout.
- We have a bug right now where we actually lose two of the three standard chrome items when vertical.
- We can’t display the target app name (unless it’s extremely short).
- We can’t display item labels (unless extremely short).
- There will be some resource usage items that we won’t be able to display, as they’ll be too long.
- We're converging on standard controls as much as possible, and the standard CommandBar does not support a vertical mode.
-

### LINKED ISSUE #3702: Remove the docking feature
### Suggested new feature or improvement

Per user feedback, this is not particularly useful, and it has a number of problems.

### Scenario

Remove the feature altogether.

### Additional details

Problems include:
- When collapsed, the only per-app information is the app’s icon + name and any resource usage items the user might have pinned to the bar. However, the icon + name are redundant when docked, and the user might not have pinned any resource usage items, plus some resource usage items will be difficult to display in collapsed vertical mode. 
- When expanded, the likelihood that docking is even feasible is dramatically reduced, given the size and position of the target window, and whether it’s maximized.
- There are many cases where docking is not feasible, including:
  - The target window is maximized or full-screen.
  - The target window is too close to the right-hand edge of the screen.
  - The target window is minimized.
  - The target app is headless.
- Even if PI successfully docked, if the user then moves/sizes/maximizes the app window, this can cause PI to undock, which increases complexity and reduces predictability. We then also have the dilemma of what to do if they later move/size/restore to a point where we could dock again – do we dock again, or would this be unexpected?
- Docking doesn’t work with desktop snapping. Windows isn’t aware of PI’s docking behavior, and PI doesn’t know when desktop snapping is happening. Consequently, when you snap an app to which you’ve docked PI, you’ll get some unworkable UX. If you tile the target app and another app, PI will be hidden. If you snap to the right-hand side of the screen, PI will be off the screen. If you snap to the bottom of the screen, there’s a good chance PI’s window will be clipped. And so on. All these behaviors will cause user frustration and confusion.
- Docking is a bug farm – there’s a lot of work to keep the PI window aligned as the user moves/sizes the target window.

### PR DESCRIPTION
**Pull request by timkur**
## Summary of the pull request

Based on the User Research Study, it was determined that the collapsed vertical bar, and the ability to snap the bar to an app window, did not resonate with as many users. Since we still have open issues for that functionary, we decided to remove it from the product for now... if UR studies come back saying folks want this, we'll invest the time to bring it back and fix the issues associated with them.
 
## Detailed description of the pull request / Additional comments

I think I got most of the code. One thing to keep in mind... originally BarWindow.cs served as a multiplexer and routed events to either horizontalbarwindow or verticalbarwindow. With this change, I removed the original BarWindow.cs and any functionality that had crept into it was moved to the horizontalbarwindow class (which was then renamed to barwindow).

## PR checklist
- [ ] Closes #3320 
- [ ] Closes #3051 
- [ ] Closes #3701 
- [ ] Closes #3702 
- [ ] Tests added/passed
- [ ] Documentation updated

### CODE REVIEW

File: tools/DevInsights/DevHome.DevInsights/Views/BarWindow.xaml
Comment (andreww-msft) [inline tools/DevInsights/DevHome.DevInsights/Views/BarWindow.xaml:1]: Looks like you're missing all the changes in https://github.com/microsoft/devhome/pull/3722/

Comment (timkur) [reply tools/DevInsights/DevHome.DevInsights/Views/BarWindow.xaml:1]: Yep, looks like your commit and my PR just missed each other. I'll pick it up when I remerge to main.

File: tools/DevInsights/DevHome.DevInsights/Strings/en-us/Resources.resw
Comment (andreww-msft) [inline tools/DevInsights/DevHome.DevInsights/Strings/en-us/Resources.resw:1]: VerticalCloseButton.ToolTipService.ToolTip #868 is also redundant, I believe.

**Review by andreww-msft — APPROVED** [review]

**Review by zadesai — APPROVED** [review]
:shipit:

File: tools/DevInsights/DevHome.DevInsights/Views/BarWindow.xaml
Comment (timkur) [reply tools/DevInsights/DevHome.DevInsights/Views/BarWindow.xaml:1]: Just did the merge.

File: tools/DevInsights/DevHome.DevInsights/Strings/en-us/Resources.resw
Comment (timkur) [reply tools/DevInsights/DevHome.DevInsights/Strings/en-us/Resources.resw:1]: Thanks. Got it!

**Pull request merged**